plugins/importexport/native/filter/NativeFilterHelper.inc.php

```diff
@@ -284,11 +378,93 @@ function parseIssueCover($filter, $node, $object) {
+						$allowedFileTypes = ['gif', 'jpg', 'png', 'webp'];
+						$extension = pathinfo(strtolower($filePath), PATHINFO_EXTENSION);
+						if (!in_array($extension, $allowedFileTypes)) {
+							$deployment->addWarning(ASSOC_TYPE_ISSUE, $object->getId(), __('plugins.importexport.common.error.invalidFileExtension'));
+							break;
+						}
+
+						$imageUrl = $n->getAttribute('src');
+						if (empty($imageUrl)) {
+							$deployment->addWarning(ASSOC_TYPE_ISSUE, $object->getId(), __('plugins.importexport.common.error.missingHref'));
+							break;
+						}
+
+						$fileContents = file_get_contents($imageUrl);
+						if ($fileContents === false) {
+							$deployment->addWarning(ASSOC_TYPE_ISSUE, $object->getId(), __('plugins.importexport.common.error.failedDownloadingFile', array('url' => $imageUrl)));
+							break;
+						}
+
+						file_put_contents($filePath, $fileContents);
+						break;
+					case 'file_url':
+						if (!$object->getCoverImage($locale)) {
+							$deployment->addWarning(ASSOC_TYPE_ISSUE, $object->getId(), __('plugins.importexport.common.error.coverImageNameUnspecified'));
+							break;
+						}
+
+						import('classes.file.PublicFileManager');
+						$publicFileManager = new PublicFileManager();
+						$filePath = $publicFileManager->getContextFilesPath($context->getId()) . '/' . $object->getCoverImage($locale);
+						$allowedFileTypes = ['gif', 'jpg', 'png', 'webp'];
+						$extension = pathinfo(strtolower($filePath), PATHINFO_EXTENSION);
+						if (!in_array($extension, $allowedFileTypes)) {
+							$deployment->addWarning(ASSOC_TYPE_ISSUE, $object->getId(), __('plugins.importexport.common.error.invalidFileExtension'));
+							break;
+						}
+
+						$imageUrl = $n->textContent;
+						if (empty($imageUrl)) {
+							$deployment->addWarning(ASSOC_TYPE_ISSUE, $object->getId(), __('plugins.importexport.common.error.missingFileUrl'));
+							break;
+						}
+
+						$fileContents = file_get_contents($imageUrl);
+						if ($fileContents === false) {
+							$deployment->addWarning(ASSOC_TYPE_ISSUE, $object->getId(), __('plugins.importexport.common.error.failedDownloadingFile', array('url' => $imageUrl)));
+							break;
+						}
+
+						file_put_contents($filePath, $fileContents);
+						break;
 					default:
 						$deployment->addWarning(ASSOC_TYPE_ISSUE, $object->getId(), __('plugins.importexport.common.error.unknownElement', array('param' => $n->tagName)));
 				}
 			}
 		}
 	}
 
+	/**
+	 * Get MIME type based on file extension
+	 * @param $fileName string
+	 * @return string MIME type
+	 */
+	protected function _getMimeTypeFromFileName($fileName) {
+		$extension = strtolower(pathinfo($fileName, PATHINFO_EXTENSION));
+		switch ($extension) {
+			case 'jpg':
+			case 'jpeg':
+				return 'image/jpeg';
+			case 'png':
+				return 'image/png';
+			case 'gif':
+				return 'image/gif';
+			case 'webp':
+				return 'image/webp';
+			default:
+				return 'application/octet-stream';
+		}
+	}
+
```

Comment:
We have such method already in the codebase, but before using it, check if the application doesn't provide it through a `getMimeType()` (or something like that) method.

```suggestion
```